### DISCUSSION TITLE
Unable to pull METplus V5.1.0 container from Dockerhub

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2305] [Installation] [answered]
I tried to pull the container on the HPC and I am getting the following response:
singularity pull docker://dtcenter/metplus
FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://dtcenter/metplus: reading manifest latest in docker.io/dtcenter/metplus: manifest unknown: manifest unknown

Do you have a resolution for this?

Thanks.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@jwraby I was able to replicate the behavior you describe through docker:
```
> docker pull dtcenter/metplus
Using default tag: latest
Error response from daemon: manifest for dtcenter/metplus:latest not found: manifest unknown: manifest unknown
```
The reason for this is that we recently removed the `latest` tag for METplus from DockerHub. When you don't specify the tag name you'd like, `docker` uses `latest` by default and that no longer exists.

The `latest` tag has lots of ambiguity and that's why we chose to remove it.

The fix is simple - just request one of the tag names that actually exists:
   https://hub.docker.com/repository/docker/dtcenter/metplus/tags

For example, [5.1.0](https://hub.docker.com/layers/dtcenter/metplus/5.1.0/images/sha256-302823f764cb19eeb890d37cb9cf8089f43ba6b9cafaea3f92fe1f3ba694d4e6?context=repo) is the most recent release and [5.1-latest](https://hub.docker.com/layers/dtcenter/metplus/5.1-latest/images/sha256-cf074e663b77e855325f0aa540666304d0212176c3582aafba2c6a4e206af23a?context=repo) is the most recent release plus all known bugfixes.

The following command DOES work fine:
```
docker pull dtcenter/metplus:5.1.0
```
So please retry your Singularity command using `docker://dtcenter/metplus:5.1.0`.

**Reply by jwraby** [reply]
John - Thanks for the quick resolution. Your command worked perfectly. Had one warning:
"destination filesystem does not support xattrs, further warnings will be suppressed"
Any idea what this means?

**Reply by JohnHalleyGotway** [reply]
I do not have an explanation for that. I'll note that I do not see that warning when pulling from DockerHub so I suspect it has something to do with the Singularity part of it. Doing some Googling of that error message, I see some additional information. But I don't have anything else to offer.

**Comment by jwraby** [comment]
Thanks John. I suspect its an issue on the HPC side. I'll contact the HPC
Help Desk. Have a great weekend!

R/

John

 

From: John Halley Gotway ***@***.***> 
Sent: Friday, August 11, 2023 11:11 AM
To: dtcenter/METplus ***@***.***>
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.***>;
Mention ***@***.***>
Subject: [URL Verdict: Neutral][Non-DoD Source] Re: [dtcenter/METplus]
Unable to pull METplus V5.1.0 container from Dockerhub (Discussion #2305)

 

I do not have an explanation for that. I'll note that I do not see that
warning when pulling from DockerHub so I suspect it has something to do with
the Singularity part of it. Doing some Googling of that error message, I see
some additional information. But I don't have anything else to offer.

-
Reply to this email directly, view it on GitHub
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fdtcenter%2FMETplus%2Fdiscussions%2F2305%23discussioncomment-6703845&dat
a=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C15966d8be1ca45581ce408db9a8e0b74%7
Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C638273707104566982%7CUnknown%7CT
WFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3
D%7C3000%7C%7C%7C&sdata=7eOMrk99JXUwvKQ99oUstajKP04NLEOLq%2B4KmGxnpok%3D&res
erved=0> , or unsubscribe
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fnotifications%2Funsubscribe-auth%2FAUX5FHZ2ELQHVUJHM4KPQH3XUZRSDANCNFSM
6AAAAAA3NAVGHU&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C15966d8be1ca4558
1ce408db9a8e0b74%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C63827370710472
3213%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1
haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=MNDkahV5EDnHFI8HURBUlg79bNMi4MCqJf
%2F7vwvEsm8%3D&reserved=0> .
You are receiving this because you were mentioned.
<https://github.com/notifications/beacon/AUX5FHZ3PMCO3OHHFKWQ5NDXUZRSDA5CNFS
M6AAAAAA3NAVGHWWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQAMZFOK.gif> Message ID:
***@***.***
***@***.***>
>

**Comment by jfrost-mo** [comment]
Issue https://github.com/dtcenter/METplus/issues/2329 tracks documenting this.

**Reply by jwraby** [reply]
Good idea. Thanks.
